agent/agent.go

```diff
@@ -257,9 +258,12 @@ func _main() int {
 
 	// Setup ENI Watcher
 	_, err = eniwatchersetup.New(ctx, state, taskEngine)
-	// TODO: Add capability/alternatives to determine task networking availability
 	if err != nil {
 		log.Errorf("Error setting up ENI Watcher: %v", err)
+		if cfg.TaskENIEnabled {
```

Comment:
We should move `_, err = eniwatchersetup.New(ctx, state, taskEngine)` under the `if cfg.TaskENIEnabled` condition.